config/config.go

```diff
@@ -109,26 +128,37 @@ type AnalyzersList struct {
 	AggregateStats   *AggregateStatsConfig   `koanf:"aggregate_stats"`
 }
 
-// NodeConfig is the configuration for chain analyzers.
-type NodeConfig struct {
-	// ChainID is the chain ID of the chain this analyzer will process.
-	// Used to identify the chain in the database.
-	ChainID string `koanf:"chain_id"`
-
-	// RPC is the node endpoint.
-	RPC string `koanf:"rpc"`
-
-	// ChainContext is the domain separation context.
-	// It uniquely identifies the chain; it is derived as a hash of the genesis data.
-	// Used as safety check to prevent accidental use of the wrong RPC endpoint.
-	ChainContext string `koanf:"chaincontext"`
+// SourceConfig has some controls about what chain we're analyzing and how to connect.
+type SourceConfig struct {
+	SingleNetwork *SingleNetworkConfig `koanf:"single_network"`
```

Comment:
> Hardcode the chain context into history.DefaultChains, with a comment and understanding that this will change as test fixtures change.

are the test chain contexts actually stable?

the other thing about history config is that the chain name is also tied to how we access sdkConfig.DefaultNetworks.

I'm going to try changing it to where SourceConfig has:

- Nodes map[string]*NodeConfig (always)
- ChainName string (when using a known chain)
- CustomChain struct (when not using a known chain)
  - History History (you just put your single network with chain context in your yaml config file)
  - SDKNetwork sdkConfig.Network (for you to provide paratime and denomination info)

Reply:
:+1: Sounds like a reasonable plan too.

To answer your questions and advocate for my suggestion a little more (but disregard if you've already gone ahead with yours):
- "are the test chain contexts actually stable?" I don't know for sure, but I assume so. The context is a hash of the genesis IIRC, and for our e2e tests, the genesis is a fixture. I guess it _might_ include something ugly like current time that changes the context every time it runs ... but I doubt it, and it IMO shouldn't.
- "chain name is also tied to how we access sdkConfig.DefaultNetworks" - I'm in favor of not using `sdkConfig.DefaultNetworks`. It only covers the current version, we already replicate a large part of it, and we'll need to replicate even more of it once we add proper support for runtimes. Just copy paste the whole thing into indexer, munge it into whatever format suits us best, and be done with it. Its contents are stable are valid forever.

Reply:
k, going with `custom_chain` for now though. let's copy over the paratime info and move away from sdkConfig.DefaultNetworks some time in the future